adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -817,32 +818,36 @@ void handlePublishedMessage(final MqttContext context) {
         }
 
         final MetricsTags.QoS qos = MetricsTags.QoS.from(ctx.message().qosLevel().value());
-        return uploadMessage(
+        final Future<TenantObject> tenantTracker = getTenantConfiguration(tenant, ctx.getTracingContext());
+
+        return tenantTracker.compose(tenantObject -> uploadMessage(
                 Objects.requireNonNull(ctx),
-                Objects.requireNonNull(tenant),
+                Objects.requireNonNull(tenantObject),
```

Comment:
this has different semantics than the check for *tenant* being `null`

Reply:
`Objects.requireNonNull(tenant)` is already being checked while retrieving the tenant object using `getTenantConfiguration(...)` and therefore its not needed again. Just I released that there is also no need for `Objects.requireNonNull(tenantObject)` inside the _compose_ block. The compose _block_ is anyway executed, if the tenant object is successfully retrieved using `getTenantConfiguration(...)`.